sklearn/metrics/_plot/roc_curve.py

```diff
@@ -386,19 +398,35 @@ def from_predictions(
         >>> X_train, X_test, y_train, y_test = train_test_split(
         ...     X, y, random_state=0)
         >>> clf = SVC(random_state=0).fit(X_train, y_train)
-        >>> y_pred = clf.decision_function(X_test)
+        >>> y_score = clf.decision_function(X_test)
         >>> RocCurveDisplay.from_predictions(
-        ...    y_test, y_pred)
+        ...    y_test, y_score)
         <...>
         >>> plt.show()
         """
+        # TODO(1.8): Remove y_pred support and use y_score only
+        if y_score is not None and not isinstance(y_pred, str):
+            raise ValueError(
```

Comment:
We need to had new tests to be sure that we properly raise this error.
Basically this is the same thing for the code non-covered in the line below.

Reply:
I added a unit test namely `test_y_score_and_y_pred_deprecation` in `test_roc_curve_display.py`. I also changed `y_pred` to `y_score` in that test file following the proposed changes in this PR.